rasa/core/brokers/broker.py

```diff
@@ -3,13 +3,25 @@
 from asyncio import AbstractEventLoop
 from typing import Any, Dict, Text, Optional, Union
 
+import aio_pika.exceptions
+import psycopg2
+import sqlalchemy.exc
+
 import rasa.shared.utils.common
 import rasa.shared.utils.io
+from rasa.shared.exceptions import ConnectionException
 from rasa.utils.endpoints import EndpointConfig
 
 logger = logging.getLogger(__name__)
 
 
+CONNECTION_ERRORS = (
+    psycopg2.OperationalError,
```

Comment:
there is no event broker using`Postgres` afaik? What about the Kafka broker?

Reply:
- what about `SQLEventBroker` that can be instantiated with `postgres` dialect ([people are using it](https://sentry.io/organizations/rasahq/issues/2009384160/?project=2801673))?
- Didn't find enough kafka related errors to ignore them here. Which errors do you suggest we catch from [this list](https://github.com/dpkp/kafka-python/blob/master/kafka/errors.py)? Not sure if we should catch the base `KafkaError`, it could hide errors from us.

Reply:
> what about SQLEventBroker that can be instantiated with postgres dialect

True 👍🏻 They definitely shouldn't use it though. If you go the extra step to use Rasa Open Source with an extra service, then they'd should use something right for the job.

Reply:
> Not sure if we should catch the base KafkaError, it could hide errors from us.

I agree. Seems too broad. How about

* `NoBrokersAvailable`
* `NodeNotReadyError`
* `KafkaProtocolError`
* `UnrecognizedBrokerVersion`
* `IncompatibleBrokerVersion`
* `AuthenticationMethodNotSupported`
* `AuthenticationFailedError`
* `TopicAuthorizationFailedError`
* `GroupAuthorizationFailedError`
* `ClusterAuthorizationFailedError`
* `UnsupportedVersionError`
*  `KafkaUnavailableError`
* `KafkaConnectionError`
* `ProtocolError`
* `KafkaConfigurationError`


Maybe `RETRY_BACKOFF_ERROR_TYPES` is also enough for now?

Haha, have you seen [this](https://github.com/dpkp/kafka-python/blob/9feeb79140ed10e3a7f2036491fc07573740c231/kafka/errors.py#L104) one? 😆

Reply:
holy crap 😄

Reply:
I think your list is better than the `RETRY_BACKOFF_ERROR_TYPES` list.

Reply:
@wochinge actually the Kafka event broker doesn't try connecting when you instantiate it, so there is no way it raises an exception: https://github.com/RasaHQ/rasa/blob/3d4f2659a6516fdac2c9cf809071cbb9abacaeda/rasa/core/brokers/kafka.py#L15-L74